src/inmanta/deploy/state.py

```diff
@@ -174,31 +175,35 @@ def add_up_to_date_resource(self, resource: ResourceIdStr, details: ResourceDeta
             self.types_per_agent[details.id.agent_name][details.id.entity_type] += 1
         self.dirty.discard(resource)
 
-    def block_resource(self, resource: ResourceIdStr, details: ResourceDetails, is_transitive: bool) -> None:
+    def block_and_update_resource(self, resource: ResourceIdStr, details: ResourceDetails) -> None:
```

Comment:
I think the name of this method should indicate that we set the status to undefined. When I read it I thought it only put blocked to `YES`